client/components/Product.js

```diff
@@ -0,0 +1,86 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { addToCart, fetchSingleProduct, deleteProduct } from "../store";
+
+
+export class SingleProduct extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+  }
+
+  componentDidMount() {
+    this.props.fetchSingleProduct(this.props.match.params.id);
+  }
+
+  render() {
+    const { product, user } = this.props;
+    return (
+      <div>
+        {
+          <div className="teainfo">
+            <img src={product.imageUrl} />
+            <div>
+              <h2>{product.name}</h2>
+              <p>{product.description}</p>
+              <h5>Quantity: { product.quantity } in stock</h5>
+
+              <button className="addtocart" onClick={() => addToCart(product)}>
+                Add To Cart
+              </button>
+            </div>
+            { user.isAdmin ? (
+              <button onClick={() => this.props.deleteProduct(product.id)}>Remove Product</button>
+            ) : null }
```

Comment:
here's a cool lil trick-
```suggestion
            { user.isAdmin && (
              <button onClick={() => this.props.deleteProduct(product.id)}>Remove Product</button>
            )}
```